hal/armv7a/zynq7000/timer.c

```diff
@@ -63,20 +61,32 @@ static int _timer_irqHandler(unsigned int n, cpu_context_t *ctx, void *arg)
 
 static time_t hal_timerCyc2Us(time_t cyc)
 {
-	return (cyc * 1000LL) / 55555LL;
+	return (cyc * 1000LL) / (time_t)(timer_common.ticksPerFreq);
 }
 
 
 static time_t hal_timerGetCyc(void)
 {
 	spinlock_ctx_t sc;
-	time_t ret;
+	time_t jiffies, cnt;
 
 	hal_spinlockSet(&timer_common.sp, &sc);
-	ret = timer_common.jiffies;
+	cnt = (time_t)(*(timer_common.ttc + cnt_value));
+	jiffies = timer_common.jiffies;
+
+	/* Check if there's pending jiffies increment */
+	if ((*(timer_common.ttc + isr) & 1) != 0) {
+		/* ISR register is clear on read, we have to update jiffies now */
+		timer_common.jiffies += timer_common.ticksPerFreq;
+
+		/* Timer might've just wrapped-around,
+		 * take counter value again */
+		jiffies = timer_common.jiffies;
+		cnt = (time_t)(*(timer_common.ttc + cnt_value));
```

Comment:
Why duplicate `cnt` and `jiffies` readout? Read their values after pending interrupt check (remove lines 82-85 and move lines 74-75 after the check).

Reply:
Because timer might've overflowed exactly between cnt readout and isr check (or might've not, we don't know). I re-read jiffies instead of incrementing mostly because of personal taste.

Reply:
Ack